diff --git a/build.rs b/build.rs
--- a/build.rs
+++ b/build.rs
@@ -243,6 +243,7 @@
 
 #[cfg(not(feature = "cmake"))]
 fn compile_zstd() {
+    let target = env::var("TARGET").unwrap_or_default();
     let mut config = cc::Build::new();
 
     // Search the following directories for C files to add to the compilation.
@@ -320,6 +321,16 @@
     // Some extra parameters
     config.include("zstd/lib/");
     config.include("zstd/lib/common");
+    if target.contains("msvc") {
+        config.include("zstd/lib/compress");
+        config.include("zstd/lib/decompress");
+        #[cfg(feature = "zdict_builder")]
+        config.include("zstd/lib/dictBuilder");
+        #[cfg(feature = "legacy")]
+        config.include("zstd/lib/legacy");
+        #[cfg(feature = "seekable")]
+        config.include("zstd/contrib/seekable_format");
+    }
     config.warnings(false);
 
     config.define("ZSTD_LIB_DEPRECATED", Some("0"));
